packages/client/bin/startRpc.ts

```diff
@@ -33,24 +33,28 @@ type RPCArgs = {
 }
 
 /**
- * Returns a jwt secret from a provided file path, otherwise saves a randomly generated one to datadir
+ * Returns a jwt secret from a provided file path, otherwise saves a randomly generated one to datadir if none already exists
  */
 function parseJwtSecret(config: Config, jwtFilePath?: string): Buffer {
-  let jwtSecret
-  if (jwtFilePath !== undefined) {
-    const jwtSecretContents = readFileSync(jwtFilePath, 'utf-8').trim()
+  let jwtSecret: Buffer
+  const defaultJwtPath = `${config.datadir}/jwtsecret`
+  if (jwtFilePath !== undefined || existsSync(defaultJwtPath)) {
+    const jwtSecretContents = readFileSync(jwtFilePath ?? defaultJwtPath, 'utf-8').trim()
     const hexPattern = new RegExp(/^(0x|0X)?(?<jwtSecret>[a-fA-F0-9]+)$/, 'g')
     const jwtSecretHex = hexPattern.exec(jwtSecretContents)?.groups?.jwtSecret
     if (jwtSecretHex === undefined || jwtSecretHex.length !== 64) {
       throw Error('Need a valid 256 bit hex encoded secret')
     }
-    config.logger.debug(`Read a hex encoded jwt secret from path=${jwtFilePath}`)
+    config.logger.debug(
+      `Read a hex encoded jwt secret from ${
+        jwtFilePath !== undefined ? `path=${jwtFilePath}` : `default path=${defaultJwtPath}`
+      }`
+    )
     jwtSecret = Buffer.from(jwtSecretHex, 'hex')
   } else {
-    jwtFilePath = `${config.datadir}/jwtsecret`
     jwtSecret = Buffer.from(Array.from({ length: 32 }, () => Math.round(Math.random() * 255)))
-    writeFileSync(jwtFilePath, jwtSecret.toString('hex'))
-    config.logger.info(`Wrote a hex encoded random jwt secret to path=${jwtFilePath}`)
+    writeFileSync(defaultJwtPath, jwtSecret.toString('hex'))
```

Comment:
Does this also write if the directory/directories do not exist?

Reply:
It does not:
```
fs.writeFileSync('abc/def/efg/out.ts', 'abc')
Uncaught Error: ENOENT: no such file or directory, open 'abc/def/efg/out.ts'
    at Object.openSync (node:fs:585:3)
    at Object.writeFileSync (node:fs:2170:35) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: 'abc/def/efg/out.ts'
```

Reply:
Oh nice catch, this means that this logic actually throws in the master branch. Will address.

Reply:
Added logic to ensure that the dir exists before writing

Reply:
Given that we here had yet another case where we were close to accidentally introduce something structurally client breaking (with the last time having a bug preventing mainnet from sync which actually made it into the release), we should really still look at increase client integration testing. This is otherwise too fragile if one needs to manually do `npm run client:start` before release for a pre-release test and this is breaking (from my historical experiences) in 15-25% of the trials (and not even catching everything like this one here).